configure.ac

```diff
@@ -3980,14 +3980,11 @@ AC_CACHE_CHECK([for flock declaration], [ac_cv_flock_decl],
     [ac_cv_flock_decl=no]
   )
 ])
-if test "x${ac_cv_flock_decl}" = xyes; then
-  AC_CHECK_FUNCS(flock,,
-    AC_CHECK_LIB(bsd,flock,
-      [AC_DEFINE(HAVE_FLOCK)
-       AC_DEFINE(FLOCK_NEEDS_LIBBSD, 1, Define if flock needs to be linked with bsd library.)
-    ])
-  )
-fi
+dnl Linking with libbsd may be necessary on AIX for flock function.
+AS_VAR_IF([ac_cv_flock_decl], [yes],
+  AC_CHECK_FUNCS([flock])
+  AC_CHECK_LIB([bsd], [flock], [FCNTL_LIBS="-lbsd"])
```

Comment:
Before this change, linking with libbsd was only checked if the check for flock failed. Now linking with libbsd is done even when flock is available without it.

May I ask, was this intentional? OpenWrt recently updated to 3.11 and we are getting a build error because of this unexpected linking (https://github.com/openwrt/packages/issues/21161). I will disable this linking on our end but I wanted to check if this should be changed here as well.

Reply:
That was an oversight. My first thought is that the `AC_CHECK_` macros should have been wrapped in a `WITH_SAVE_ENV()` macro.

Reply:
Thanks for the report.

Reply:
See gh-105236